src/libraries/System.Security.Cryptography/tests/X509Certificates/X509Certificate2PemTests.cs

```diff
@@ -444,6 +444,133 @@ public static void CreateFromEncryptedPem_MLKem_Pkcs8_Success()
             }
         }
 
+        [ConditionalFact(typeof(MLDsa), nameof(MLDsa.IsSupported))]
+        public static void CreateFromPem_MLDsa_Pkcs8_Success()
+        {
+            (string CertificatePem, string PrivateKeyPem, string Thumbprint)[] cases =
+            [
+                (
+                    MLDsaTestsData.IetfMLDsa44.CertificatePem,
+                    MLDsaTestsData.IetfMLDsa44.PrivateKeyPem_Seed,
+                    "9762DDD44288AF89BDE9213F63212E273815EBBE37F2CE918C496365B1382165"
+                ),
+                (
+                    MLDsaTestsData.IetfMLDsa44.CertificatePem,
+                    MLDsaTestsData.IetfMLDsa44.PrivateKeyPem_Expanded,
+                    "9762DDD44288AF89BDE9213F63212E273815EBBE37F2CE918C496365B1382165"
+                ),
+                (
+                    MLDsaTestsData.IetfMLDsa44.CertificatePem,
+                    MLDsaTestsData.IetfMLDsa44.PrivateKeyPem_Both,
+                    "9762DDD44288AF89BDE9213F63212E273815EBBE37F2CE918C496365B1382165"
+                ),
+                (
+                    MLDsaTestsData.IetfMLDsa65.CertificatePem,
+                    MLDsaTestsData.IetfMLDsa65.PrivateKeyPem_Seed,
+                    "7BF5DA6C6EE25C59ABB6B3561C27092872FEC9052BCB98A5AA94DFA806057BD2"
+                ),
+                (
+                    MLDsaTestsData.IetfMLDsa65.CertificatePem,
+                    MLDsaTestsData.IetfMLDsa65.PrivateKeyPem_Expanded,
+                    "7BF5DA6C6EE25C59ABB6B3561C27092872FEC9052BCB98A5AA94DFA806057BD2"
+                ),
+                (
+                    MLDsaTestsData.IetfMLDsa65.CertificatePem,
+                    MLDsaTestsData.IetfMLDsa65.PrivateKeyPem_Both,
+                    "7BF5DA6C6EE25C59ABB6B3561C27092872FEC9052BCB98A5AA94DFA806057BD2"
+                ),
+                (
+                    MLDsaTestsData.IetfMLDsa87.CertificatePem,
+                    MLDsaTestsData.IetfMLDsa87.PrivateKeyPem_Seed,
+                    "EE6DB12F8B68B814A9FF912F6BADB8299EA9BAEBF121FCEFFA60CF23D3B97E34"
+                ),
+                (
+                    MLDsaTestsData.IetfMLDsa87.CertificatePem,
+                    MLDsaTestsData.IetfMLDsa87.PrivateKeyPem_Expanded,
+                    "EE6DB12F8B68B814A9FF912F6BADB8299EA9BAEBF121FCEFFA60CF23D3B97E34"
+                ),
+                (
+                    MLDsaTestsData.IetfMLDsa87.CertificatePem,
+                    MLDsaTestsData.IetfMLDsa87.PrivateKeyPem_Both,
+                    "EE6DB12F8B68B814A9FF912F6BADB8299EA9BAEBF121FCEFFA60CF23D3B97E34"
+                ),
+            ];
+
+            foreach ((string CertificatePem, string PrivateKeyPem, string Thumbprint) in cases)
```

Comment:
since this is deconstructing, rather than declaring a tuple shape, the names should all be camelCase, not pascalCase.

```suggestion
            foreach ((string certificatePem, string privateKeyPem, string thumbprint) in cases)
```